### PR TITLE
[blog] Automate Your Infrastructure with Automation API and C#

### PR DESCRIPTION
**Pull request by orionstudt**

### CODE REVIEW

File: content/blog/automation-api-dotnet/index.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+title: "Automate Your Infrastructure with Automation API and C#"
+date: tbd
+meta_desc: "tbd"
+meta_image: automation_api.png
+authors:
+- joshua-studt
```
Comment (leezen) [inline content/blog/automation-api-dotnet/index.md:7]: FYI: you'll want to add a `.toml` file to `data/team/team` for this to render correctly.

**Comment by spara** [comment]
Josh, I manage the blog, let me know if you need any assistance, always happy to help out.

sophia

**Comment by orionstudt** [comment]
Hi @spara! I'd love an assist. I can put some more energy into this when I get off work today buy when I started this I was looking primarily at the [python announcement post](https://www.pulumi.com/blog/automation-api-python/) as an example. That post doesn't really talk about python implementation specifics at all - the only thing python is the code samples. It seems to mostly be a post that describes the Automation API itself, which made it a difficult example to follow since I didn't want to just copy your homework!

So, how much of this post do we want to be reiterating what the Automation API is and what it does? I was just going to get right into dotnet examples after linking to a more thorough post describing the feature, but that will make this post significantly more brief.

**Comment by spara** [comment]
Go ahead and write up the dotnet examples part and I can throw in a paragraph about Auto API and some fresh new use cases like our new Github Actions provider. I can look at your examples and reference those as well.

File: content/blog/automation-api-dotnet/index.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+---
+title: "Automate Your Infrastructure with Automation API and C#"
+date: tbd
+meta_desc: "tbd"
```
Comment (spara) [inline content/blog/automation-api-dotnet/index.md:4]:
```suggestion
meta_desc: "C# developers can programmatically build infrastructure (with out a CLI) using the Pulumi Automation API package. "
```

File: content/blog/automation-api-dotnet/index.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+await green.UpAsync();
+
+// do your cutover to "green" logic
+// and then teardown "blue"
+
+var blue = await WorkspaceStack.SelectAsync("blue", workspace);
+await blue.DestroyAsync();
+```
+
+See the full Automation API examples in [C#](https://github.com/pulumi/automation-api-examples/tree/main/dotnet).
```
Comment (spara) [inline content/blog/automation-api-dotnet/index.md:71]:
```suggestion
See full Automation API examples in [C#](https://github.com/pulumi/automation-api-examples/tree/main/dotnet).
```

**Review by spara — CHANGES_REQUESTED** [review]
Made a few changes, would like @leezen and @infin8x to also review

**Review by leezen — APPROVED** [review]
Looks great! I just had a couple small things I didn't have a huge opinion on and defer to @orionstudt

File: content/blog/automation-api-dotnet/index.md [hunk 4]
```diff
@@ -13,11 +13,15 @@ tags:
...
 
 ## Using Automation API in .NET
 
-The `Pulumi.Automation` [NuGet package](https://github.com/pulumi/pulumi/tree/master/sdk/dotnet/Pulumi.Automation) exposes a `LocalWorkspace` that empowers you to create and manage [Stacks]({{< relref "/docs/intro/concepts/stack" >}}), as well as a `WorkspaceStack` that is your programmatic representation of a Stack use to perform infrastructure updates, refresh, previews, and destroy. The Automation API makes it trivial to run Pulumi programs inline:
+The `Pulumi.Automation` [NuGet package](https://github.com/pulumi/pulumi/tree/master/sdk/dotnet/Pulumi.Automation) exposes a `LocalWorkspace` for  creating and managing Pulumi [Stacks]({{< relref "/docs/intro/concepts/stack" >}}), and a `WorkspaceStack` that is a programmatic representation of a Stack for updating, refreshing, previewing, and destroying cloud resources. The Automation API makes it trivial to run Pulumi programs inline:
```
[the remark this answers is not in the document]
Comment (orionstudt) [reply content/blog/automation-api-dotnet/index.md]: Yea that sounds better - done

**Comment by orionstudt** [comment]
Should be good now I think then!

File: content/blog/automation-api-dotnet/index.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+- Automation API
+- C#
+- csharp
+- dotnet
+- .NET
+---
+
+{{% notes type="info" %}}
+Joshua Studt is a Solutions Architect at Financial Independence Group and a Pulumi Community member who contributed the C# package for Automation API.
+{{% /notes %}
```
Comment (spara) [inline content/blog/automation-api-dotnet/index.md:18]:
```suggestion
{{% /notes %}}

```

File: content/blog/automation-api-dotnet/index.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
+---
+title: "Automate Your Infrastructure with Automation API and C#"
+date: tbd
```
Comment (spara) [inline content/blog/automation-api-dotnet/index.md:3]:
```suggestion
date: 2021-03-08
```

File: content/blog/automation-api-dotnet/index.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+{{% notes type="info" %}}
+Joshua Studt is a Solutions Architect at Financial Independence Group and a Pulumi Community member who contributed the C# package for Automation API.
+{{% /notes %}
+Currently available in public preview, Pulumi's Automation API lets you to programmatically call upon the Pulumi engine to provision your infrastructure. Today, we are excited to announce C# support for Automation API, enabling .NET developers to automate infrastructure deployments, create complex orchestration workflows, build custom ops tooling, and build cloud frameworks. Read more about the Automation API [here]({{< relref "/blog/automation-api" >}}).
+
+## Using Automation API in .NET
+
+The `Pulumi.Automation` [NuGet package](https://www.nuget.org/packages/Pulumi.Automation) exposes a `LocalWorkspace` for  creating and managing Pulumi [Stacks]({{< relref "/docs/intro/concepts/stack" >}}), and a `WorkspaceStack` that is a programmatic representation of a Stack for updating, refreshing, previewing, and destroying cloud resources. The Automation API makes it trivial to run Pulumi programs inline:
+
+
```
Comment (spara) [inline content/blog/automation-api-dotnet/index.md:25]:
```suggestion
```

File: content/blog/automation-api-dotnet/index.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+await green.UpAsync();
+
+// do your cutover to "green" logic
+// and then teardown "blue"
+
+var blue = await WorkspaceStack.SelectAsync("blue", workspace);
+await blue.DestroyAsync();
+```
+
+See full Automation API examples in [C#](https://github.com/pulumi/automation-api-examples/tree/main/dotnet).
```
Comment (spara) [inline content/blog/automation-api-dotnet/index.md:75]:
```suggestion
See full Automation API examples in [C#](https://github.com/pulumi/automation-api-examples/tree/main/dotnet). Let us know if you're using Automation API on [Twitter](https://twitter.com/PulumiCorp) or in [Pulumi Comunity Slack](https://slack.pulumi.com/).
```

**Review by spara — APPROVED** [review]
Fixed shortcode error, removed extraneous line, set publish date, and added call to action.

File: content/blog/automation-api-dotnet/index.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+- csharp
+- dotnet
+- .NET
+---
+
+{{% notes type="info" %}}
+Joshua Studt is a Solutions Architect at Financial Independence Group and a Pulumi Community member who contributed the C# package for Automation API.
+{{% /notes %}}
+
+Currently available in public preview, Pulumi's Automation API lets you to programmatically call upon the Pulumi engine to provision your infrastructure. Today, we are excited to announce C# support for Automation API, enabling .NET developers to automate infrastructure deployments, create complex orchestration workflows, build custom ops tooling, and build cloud frameworks. Read more about the Automation API [here]({{< relref "/blog/automation-api" >}}).
```
Comment (infin8x) [inline content/blog/automation-api-dotnet/index.md:21]:
```suggestion
Currently available in public preview, Pulumi's Automation API enables you to provision your infrastructure programmatically using the Pulumi engine. Today, we are excited to announce C# support for Automation API, enabling .NET developers to automate infrastructure deployments, create complex orchestration workflows, build custom ops tooling, and build cloud frameworks. Read more about the Automation API [here]({{< relref "/blog/automation-api" >}}).
```

File: content/blog/automation-api-dotnet/index.md [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+await green.UpAsync();
+
+// do your cutover to "green" logic
+// and then teardown "blue"
+
+var blue = await WorkspaceStack.SelectAsync("blue", workspace);
+await blue.DestroyAsync();
+```
+
+See full Automation API examples in [C#](https://github.com/pulumi/automation-api-examples/tree/main/dotnet). Let us know if you're using Automation API on [Twitter](https://twitter.com/PulumiCorp) or in [Pulumi Comunity Slack](https://slack.pulumi.com/).
```
Comment (infin8x) [inline content/blog/automation-api-dotnet/index.md:76]:
```suggestion
See full Automation API examples in [C#](https://github.com/pulumi/automation-api-examples/tree/main/dotnet). Let us know if you're using Automation API on [Twitter](https://twitter.com/PulumiCorp) or in our [Community Slack](https://slack.pulumi.com/).
```

**Review by infin8x — APPROVED** [review]
This looks great - thanks @orionstudt for contributing! A few suggestions but otherwise LGTM.

**Pull request merged**